### PR TITLE
Add im2col function for Chainer Variable

### PR DESCRIPTION
**Pull request by yuyu2172**
This PR adds im2col that acts on chainer.Variable.

### CODE REVIEW

File: chainer/functions/array/im2col.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    Patches are extracted at positions shifted by multiples of ``stride`` from
+    the first position ``-pad`` for each spatial axis.
+    The right-most (or bottom-most) patches do not run over the padded spatial
+    size.
+
+    Notation: here is a notation for dimensionalities.
+
+    - :math:`n` is the batch size.
+    - :math:`c` are the number of the input channels.
```
Comment (muupan) [inline chainer/functions/array/im2col.py]: s/are/is/

File: chainer/functions/array/im2col.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                x, self.kh, self.kw, self.sy, self.sx, self.ph, self.pw,
+                cover_all=self.cover_all, dy=self.dy, dx=self.dx)
+        n, c, kh, kw, out_h, out_w = y.shape
+        y = y.reshape(n, c * kh * kw, out_h, out_w)
+        return y,
+
+    def backward(self, inputs, grad_outputs):
+        x, = inputs
+        xp = cuda.get_array_module(x)
+        _, _, h, w = x.shape
```
Comment (muupan) [inline chainer/functions/array/im2col.py]: This line is not used.

File: chainer/functions/array/im2col.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    - :math:`s_Y` and :math:`s_X` are the stride of the filter.
+    - :math:`p_H` and :math:`p_W` are the spatial padding size.
+
+    The output size :math:`(h_O, w_O)` is determined by the following
+    equations:
+
+    .. math::
+
+       h_O &= (h + 2p_H - k_H) / s_Y + 1,\\\\
+       w_O &= (w + 2p_W - k_W) / s_X + 1.
```
Comment (muupan) [inline chainer/functions/array/im2col.py]: It would be better to consider `dilate` and `cover_all` in equations.

**Comment by yuyu2172** [comment]
Thank you very much for reviewing.
I fixed what you pointed as well as some grammatical issues in the documentation.

**Comment by mitmul** [comment]
LGTM

**Pull request merged**